src/main/java/edu/ucdavis/dss/ipa/api/deserializers/SectionGroupCostDeserializer.java

```diff
@@ -114,6 +71,18 @@ public Object deserialize(JsonParser jsonParser, DeserializationContext deserial
             sectionGroupCost.setOriginalInstructor(originalInstructor);
         }
 
+        if (node.has("budgetScenarioId")) {
+            BudgetScenario budgetScenario = new BudgetScenario();
```

Comment:
Is it safe to assume the budget scenario is new in the deserializer?

Reply:
Its not that we assume the scenario is new, this is the pattern we are using to translate the view DTO (that is flattened with transient id values) back into nested resources.